js/plates-model/model-worker.js

```diff
@@ -97,6 +98,14 @@ onmessage = function modelWorkerMsgHandler (event) {
   } else if (data.type === 'restoreInitialSnapshot') {
     self.m = model = Model.deserialize(initialSnapshot)
     snapshots.length = 0
+  } else if (data.type === 'takeLabeledSnapshot') {
```

Comment:
I went back and forth on this, but I settled on making a new worker call for taking a labeled snapshot. I considered implementing a single call that would pick a storage object based on whether or not a label was passed...but the storage and restoring works so differently that I didn't want to overload things.